File: javascript/src/models/bigQueryPatchConfig.ts
// this file is @generated

export interface BigQueryPatchConfig {
  projectId?: string;
  datasetId?: string;
  tableId?: string;
  credentials?: string;
}

export const BigQueryPatchConfigSerializer = {
  _fromJsonObject(object: any): BigQueryPatchConfig {
    return {
      projectId: object["projectId"],
      datasetId: object["datasetId"],
      tableId: object["tableId"],
      credentials: object["credentials"],
    };
  },

  _toJsonObject(self: BigQueryPatchConfig): any {
    return {
      projectId: self.projectId,
      datasetId: self.datasetId,
      tableId: self.tableId,
      credentials: self.credentials,
    };
  },
};

File: rust/src/models/big_query_patch_config.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct BigQueryPatchConfig {
    #[serde(rename = "projectId")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub project_id: Option<String>,

    #[serde(rename = "datasetId")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub dataset_id: Option<String>,

    #[serde(rename = "tableId")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub table_id: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub credentials: Option<String>,
}

impl BigQueryPatchConfig {
    pub fn new() -> Self {
        Self {
            project_id: None,
            dataset_id: None,
            table_id: None,
            credentials: None,
        }
    }
}

impl Default for BigQueryPatchConfig {
    fn default() -> Self {
        Self::new()
    }
}
